### PR TITLE
src layout.

### PR DESCRIPTION
**Pull request by RubelMozumder**

### CODE REVIEW

File: .github/workflows/pylint.yaml
Comment (mkuehbach) [inline .github/workflows/pylint.yaml:1]: Recommend to use the newer github actions from our pynxtools-PLUGINS to test for all python version automatically

**Review by mkuehbach — CHANGES_REQUESTED** [review]
functionally, lgtm but recommendation to check CI for all python version 3.8 - 3.12

**Comment by RubelMozumder** [comment]
@mkuehbach this should be fine now.

**Review by lukaspie — APPROVED** [review]
```
[tool.ruff]
include = ["nyaml/*.py", "tests/*.py"]
```
should be changed to
```
[tool.ruff]
include = ["src/nyaml/*.py", "tests/*.py"]
```

Other than that, LGTM.

**Pull request merged**